The first version of cachi2 website

slimreaper35 opened the pull request:
To try anything locally run: `mkdocs serve`

# Maintainers will complete the following section

- [ ] Commit messages are descriptive enough
- [ ] Code coverage from testing does not decrease and new code is covered
- [ ] Docs updated (if applicable)
- [ ] Docs links in the code are still valid (if docs were updated)

**Note:** if the contribution is external (not from an organization member), the CI
pipeline will not run automatically. After verifying that the CI is safe to run:

- [approve GitHub Actions workflows][approve-gh-actions] by clicking a button
- approve the Red Hat Trusted App Pipeline container build by commenting `/ok-to-test`
  (as is the [standard for Pipelines as Code][pac-running-pipeline])

[approve-gh-actions]: https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks
[pac-running-pipeline]: https://pipelinesascode.com/docs/guide/running/#running-the-pipelinerun

slimreaper35 commented:
preview from my own fork: https://cachi2.readthedocs.io

slimreaper35 commented on docs/index.md:
```diff
@@ -0,0 +1,3 @@
+# Cachi2 documentation
+
+TODO
```
We could, but the content must be inside `docs`  directory, so we would have a symlink to README.md in there.

slimreaper35 replied:
(similarly to other documents like CONTRIBUTING.md)

brunoapimentel replied:
The reason I mentioned it is because it feels like the README content is what could be our homepage, since it is a docs website.

slimreaper35 replied:
Also, we will have to do something with links to other documents because some of them might not be valid on the website.

slimreaper35 commented:
- rebased to include cargo in the website
- added [permalinks](https://www.mkdocs.org/user-guide/writing-your-docs/#linking-to-pages) extension
- replaced default material theme logo with lock.svg (lock the chef in the kitchen... :)

taylormadore reviewed (commented):
Nice job to start. I think the home page is really the only major piece missing here for phase 1

ben-alkov commented on mkdocs.yml:
```diff
@@ -0,0 +1,54 @@
+# Navigation bar
+nav:
+  - Home: index.md
+  - Usage: usage.md
```
I feel like the strict "Usage" section isn't overly long, but the examples make it very long. Perhaps we could split them out into a new usage_examples.md?

taylormadore replied:
Yes, sorry that was what I meant. Thanks for clarifying

slimreaper35 replied:
> I feel like the strict "Usage" section isn't overly long, but the examples make it very long. Perhaps we could split them out into a new usage_examples.md?

True. Actually, there are missing examples for recently added package managers in usage.md, like yarn v1, bundler, cargo. Those, are separate sections in their standalone documents. (It's a big mess, in my opinion).

slimreaper35 commented on requirements-extras.txt:
```diff
@@ -4,88 +4,88 @@ aiohappyeyeballs==2.4.6 \
-    --hash=sha256:f6a19bcab7fbd8f8649d6595624856635159a6527861b9cdc3447af288a00c00 \
-    --hash=sha256:f752e80606b132140883bb262a457c475d219d7163d996dc9072434ffb0784c4 \
-    --hash=sha256:f7914ab70d2ee8ab91c13e5402122edbc77821c66d2758abb53aabe87f013287
+aiohttp==3.11.13 \
```
No, I just run `nox -s pip-compile`

slimreaper35 replied:
So it's because `--refresh` [option](https://github.com/hermetoproject/cachi2/blob/main/noxfile.py#L153).

slimreaper35 commented:
> Nice job to start. I think the home page is really the only major piece missing here for phase 1

Thanks. Do you have an idea what to put there?

slimreaper35 commented:
GitHub page from my fork -> https://slimreaper35.github.io/cachi2/

a-ovchinnikov commented on mkdocs.yml:
```diff
@@ -0,0 +1,84 @@
+repo_url: https://github.com/hermetoproject/cachi2
+edit_uri: edit/main/docs
+
+# Feedback gathering
```
Do we really need or want feedback to the docs? Do we plan to collect it and act on it? This is an open-source project oriented on developers, if someone is not happy with a page they should probably submit a PR or at least an issue.

slimreaper35 replied:
That's optional. We can incorporate it later.

a-ovchinnikov replied:
I am not against it, I have already approved the change, but I am genuinely interested if we really want this and of how this is supposed to work.

slimreaper35 replied:
Ok, removed.

slimreaper35 replied:
Also, removed the logo, which is not necessary for now...I was just testing it out.
